python/spec/dependabot/python/version_spec.rb

```diff
@@ -70,126 +75,46 @@
   end
 
   describe "#<=>" do
-    subject { version <=> other_version }
-
-    context "compared to a Gem::Version" do
-      context "that is lower" do
-        let(:other_version) { Gem::Version.new("0.9.0") }
-        it { is_expected.to eq(1) }
-      end
-
-      context "that is equal" do
-        let(:other_version) { Gem::Version.new("1.0.0") }
-        it { is_expected.to eq(0) }
-
-        context "but our version has a local version" do
-          let(:version_string) { "1.0.0+gc.1" }
-          it { is_expected.to eq(1) }
-        end
-
-        context "but our version has a v-prefix" do
-          let(:version_string) { "v1.0.0" }
-          it { is_expected.to eq(0) }
-        end
-      end
-
-      context "that is greater" do
-        let(:other_version) { Gem::Version.new("1.1.0") }
-        it { is_expected.to eq(-1) }
+    sorted_versions = [
+      "",
+      "0.9",
+      "1.0.0-alpha",
+      "1.0.0-a.1",
+      "1.0.0-beta",
+      "1.0.0-b.2",
+      "1.0.0-beta.11",
+      "1.0.0-rc.1",
+      "1",
+      # "1.0.0.post", TODO fails comparing to 1
```

Comment:
The more thorough testing is showing here that dependabot thinks `1 > 1.0.0.post`, but yet `1 < 1.0.0.post1` so I think there's a bug here. 

According to [PEP 440](https://peps.python.org/pep-0440/#post-releases), post release identifiers are for non-material changes like fixing release notes. So I assume if we fix that then dependabot will be bumping a lot of major versions to post release versions.

But I want to look into this more after we fix epoch support so I'll make a followup issue.